### DISCUSSION TITLE
Koriki for RG35XX Beta 0.7.2

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#3] [Announcements]
# Introduction
This is the second public release of Koriki for the RG35XX. Note that this is an beta release, so while the system is usable there may be issues that still need to be solved.

# Installation:
* Flash the koriki-rg35xx image to a SDCARD using balena etcher or a similar software
* Boot with the SDCARD. The first time the firmware will expand the second partition and copy some firmware files to that partition
* Important: this version now uses FAT32 for the second partition, so it should be accessible on Mac/PC without any dedicated software
* If you use a second SDCARD, the system will install those files in the last partition of the second card
* Note that the first time the system boots it will take a bit longer to expand the card and copy files over. Even when the menu is up, there will be a few seconds when the system will be a bit slow because it's adding the virtual memory image
* After the first boot the system will create some configuration folders as well as the roms and bios folders. Add your roms to the roms folder, and your bios to the roms folder.

# Update from previous installations:

* NOTE: If you installed v0.7 or higher, it's safe to use the update file. If you installed v0.6 or a previous version, you should install the full image instead since the partition tables have changed.
* Uncompress the file boot.tar.xz, and replace the contents of the first partition of your SDCARD image with this one (you should see files like uImage, ramdisk.img, boot folder, etc.)
* If you run into issues with content not starting or any other issue, rename your system folder (from the last partition of your main SDCARD or your second SDCARD), and start again.

# IMPORTANT
If you use a second SDCARD that you have previously used with koriki for RG35XX, please rename your system folder, and start again.

# Charger behavior:
* The charging logic is slightly different from the one in the stock firmware (or Garlic). The system reimplements the charger logic that is performed via software and not via hardware in this system.
* Shutdown:
  * If the charger power cable is connected to the USB C port, once the system is shutdown it will boot into the charger mode. You will briefly see a battery symbol and the system will seem to be off but the charger led will be on (orange). To boot from this state, briefly press power key.
  * If the charger cable is not connected, the system will be off completely. To boot again from this state you need to press the power button a few seconds and the system will boot.
* Reboot:
  * The system will reboot as expected
 
# ChangeLog:
2023/06/13
* Added power button shutdown. Short press power button will shutdown the console. If the charger is connected it will boot into charger mode. From there another short press boots the console
* Fixed shutdown while charging
* Added brightness control shortcut with ``MENU``+``VOLUME+/VOLUME-``
* Fixed Sega32X launcher
* Changed AmstradCPC core to cap32

2023/06/12
* Update to fix an issue with the swap and share partitions being overwritten
* Reapplied the SDL patch that was not correctly applied (thanks again @johnnyonflame)
* Fixed typo in mastersystem path
* Added proper openbor launchers
* Added o2em launcher
* Fixed sega32x path

2023/06/11
* Added SNES mednafen-supafaust core (default for SNES). StarFox now runs close to 60fps without overclock or video threading
* Added MAME km_mame_2003_xtreme that runs much better for some roms (default for MAME)
* Fixed swap virtual memory generation, it has its own dedicated partition now, should be much more efficient and not impact FAT32 partitions
* Added additional ports location
* Added SDL VT patch to solve some issues with some ports not returning to simplemenu (thanks @johnnyonflame)

2023/06/08
* Fixed simplemenu NO GAMES FOUND message. Now it will go back to the last selection after a game is closed or after restart
* Changed data partition to FAT32 format
* Added controller configuration for standalone emulators (flycast, PPSSPP, mupen64plus, and Drastic)
* Changed default emulators, removed some missing emulators from the core list

2023/06/06
* First public release (alpha v0.5)
* Implemented global audio control (thanks to [adixal](https://codeberg.org/adixal) for his muOS joyaudio driver)
* Improved the charger logic. System now should properly respond to shutdown or reboot commands in koriki/simplemenu. Just go to settings -> Session, press left/right until you see either shutdown or reboot and press A to confirm. See charger behavior below for details.
* Updated flycast and ppsspp cores to the latest release
* Added drastic, flycast, and ppsspp application launchers. 

2023/06/04
* Updated mgba core
* Added cores: vba-m, gpsp, stella2014
* Added flycast standalone
* Fixed partition autoexpand

2023/06/03
* Updated launchers and menu sections so most systems should now work right out of the box
* Fixed charger led indicators
* Added od-commander
* Added missing flycast_libretro cores
* Added virtual memory back (should help to launch N64 titles)

2023/06/02
* Updated ``koriki-rg35xx_20230602.img.gz``
* Fixed battery charge indicator

2023/06/01
* Reworked system settings menu
* Added system overclock support
* Added cores back to the image
* Added battery charger logic (use reboot or shutdown in simplemenu to change behavior)
* System should work with either 1 sdcard or 2 sdcard (SD1 koriki, SD2 roms)
  * By default roms use the batocera roms names and assume those are located in ``/userdata/roms/<system>``
  * You can use garlic or OFW images for the roms in the second SDCARD slot, but you will need to modify the paths in ``/userdata/system/.simplemenu/sections_groups``
* SDCard partition auto expand is current broken

2023/05/31
* ~~Added koriki_share_rg35xx.zip archive. This archive is intended to be used for a second sdcard. Uncompress at the top of the archive.~~
* It contains the cores, as well as the folder structure to add roms to the system

2023/05/30
* First development release
* Image automatically generated
* SDCARD expansion
* External SDCARD compatibility
* Only a few set of cores included

## Working:
* Navigation, rom selection, launching roms
* Brightness control
* Battery indicator
* Initial volume control via keys (menu does not reflect control change)
* Volume control for all emulators (thank to [adixal](https://codeberg.org/adixal) for this)
* System overclock control
* Retroarch menu can be opened with the ``Menu`` key. Quit with ``START``+``SELECT``
* Content auto-copied when a new SDCARD is inserted
* Auto mounts last partition of a second sdcard if present

## Not working:
* ~~By default the system will start showing a message saying "No Games Found". Just press B to go back and select your rom. This is a bug that needs to be fixed in this version of simplemenu~~
* Some menus, settings are still incomplete
* Some emulators need to be updated
* ~~Standalone emulators require key mappings to be updated~~
* Input event issues with retroarch (map retroarch configuration manually)
* Application launchers (e.g. Wifi, need to be updated)
* ~~When the console starts or the menu is opened after a game is closed simplemenu shows there are no games available. Press B to go back and then you can go into the selection.~~
* ~~Controls for standalone emulators are not properly mapped~~
* Networking is still not configured
* External controllers are not working at this point (kernel is in adb mode by default)
* Selecting a core in simplemenu changes the background color (mostly cosmetic)

## Tweaking configuration:
* HOME is ``/userdata/system`` (also alias to /mnt/SDCARD)
* Simplemenu files are in ``$HOME/.simplemenu``
* Sections are in ``$HOME/.simplemenu/section_groups`` (self explanatory)
* Emulator launchers are in ``$HOME/.simplemenu/launchers`` (self explanatory, but look for the ones like neogeo_libretro that have been already adapted) 
* By default the firmware expects batocera-like roms partitions (with folders under roms like neogeo, mame, megadrive, etc.). It's possible add additional roms paths in ``$HOME/.simplemenu/section_groups`` .ini files. For instance, for master system you can see an example of batocera path (/userdata/roms/mastersystem) and Miyoo Koriki path (/userdata/Roms/MS). 
```[MASTER SYSTEM]
execs = /userdata/system/.simplemenu/launchers/picodrive_libretro,/userdata/system/.simplemenu/launchers/genesis_plus_gx_libretro,/userdata/system/.simplemenu/launchers/gearsystem_libretr
romDirs = /userdata/roms/mastersystem/,/userdata/Roms/MS
romExts = .zip,.sms,.7z
```
* The snapshots/art/thumbnail previews are by default under ``media/images`` (e.g. neogeo/media/images/mslug.png). You can change that path in $HOME/.simplemenu/config.ini and change the value of ``media_folder = media/images`` (for instance ``media_folder = Img`` is the default Koriki/miyoo path).

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/'>Koriki for RG35XX Beta 0.7.2</a>.</em>

**Comment by DeLaBath-33** [comment]
@acmeplus Do you know how to run ports games?
I have tried Duke Nukem port from .rg35xx.com web site but it does not run. 
Maybe they are only compatible with GarlicOS, I don't know.
Thanks for your help and for your project !

**Reply by cyberbol** [reply]
I wish to know how to run New super mario bron from NDS ?

**Reply by acmeplus** [reply]
The ports from rg35xx.com are not compatible. Those are compiled for a different architecture. Several folks on discord have ported their ports to the batocera/koriki architecture. Check https://discord.com/channels/741895796315914271/1086371389965078663

**Reply by acmeplus** [reply]
For NDS games you just need to drop then in the roms/nds folder. If the roms are compatible with Drastic they should start without issues.

**Reply by DeLaBath-33** [reply]
@acmeplus thanks for your answer. 
Could you please invite me to this discord channel?

**Reply by MauricioSuporte** [reply]
will we have any language change function in the system? I can help with the translation into my language

**Reply by acmeplus** [reply]
@DeLaBath-33 this is the invite link: https://discord.gg/retrohandhelds look for rg35xx and then batocera under that.

**Reply by DeLaBath-33** [reply]
Thanks 🙏

**Comment by JorgeMtz** [comment]
@acmeplus How long does it take to boot the first time?, i installed this version on a new sd and i left it for like 40 min, came back and it was still on the boot screen :(, should i install a previus version or this version should work fine? i ended up changing the sd with the stock firmware to keep playing

**Reply by JorgeMtz** [reply]
btw i had previusly installed Batocera Beta v0.8.1 and it worked fine, koriki for some reason is not working for me :(

**Reply by acmeplus** [reply]
That's odd, koriki uses basically the same kernel, but not an exact copy. Koriki should get the SDCARD expanded relatively quickly (32GB is just 10-15s). Then it copies the content from the internal data to the SHARE partition. After that first time koriki cold boots in about 15s.
If you don't get it to boot, try to copy the uImage and kernel.dtb from batocera 0.8.1 to the first partition of koriki and try again. Let me know how that goes.

**Reply by cyberbol** [reply]
Hi
 I also wanted use a koriki very much, but it's start only once. First time.  After that when I unplug sd card to copy any roms in it, plug again and koriki not start anymore.  Just flash logo wallpaper and then black screen.
I tried 3 times step by step all. Process and always same.
I'm.using Kingston 128gb card

**Reply by JorgeMtz** [reply]
@acmeplus it worked, thank you 😄, btw @cyberbol that happen to me, when you turn on the device and you get the black screen just do a quick press on the power button and it will start, at least that worked for me

**Reply by cyberbol** [reply]
Will try. Do you also did what acmeplus asked you?. Did you copy files from batocera?

**Reply by cyberbol** [reply]
I just really want to manage to run New Super Mario Bros on. NDS

**Reply by JorgeMtz** [reply]
yes, for me it didn't work just flashing this img to the sd card, it was stuck on the boot image, but i had another sd with the Batocera Beta v0.8.1, just copied the 2 files acmeplus told me into the koriki sd card and it started working

**Reply by acmeplus** [reply]
For those of you having the same boot issue, you can download the archive now present in the release page: https://github.com/rg35xx-cfw/Koriki/releases/download/koriki_rg35xx_v0.7.3/bootfix_for_some_revisions.zip

**Comment by cyberbol** [comment]
Great. Did you try maybe New Super Mario Bros NDS on this koriki?

**Reply by DeLaBath-33** [reply]
Yes it works fine.
